components/NavigationDrawer/src/private/MDCBottomDrawerContainerViewController.m

```diff
@@ -481,6 +508,91 @@ - (void)preferredContentSizeDidChangeForChildContentContainer:(id<UIContentConta
   [self.view setNeedsLayout];
 }
 
+- (void)animateToPreferredContentHeight:(CGFloat)preferredContentHeight
+                           withDuration:(NSTimeInterval)duration
+                             completion:(void (^_Nullable)(BOOL))completion {
+  CGSize newPreferredContentSize =
+      CGSizeMake(CGRectGetWidth(self.view.bounds), preferredContentHeight);
+  if ([self isMobileLandscape]) {
+    self.contentViewController.preferredContentSize = newPreferredContentSize;
+    return;
+  }
+  CGPoint contentOffset =
+      [self calculateContentOffsetWithPreferredContentHeight:preferredContentHeight];
+  CGFloat precentageOfFullScreen =
+      [self precentageOfFullScreenWithPreferredContentHeight:preferredContentHeight];
+  BOOL shouldSetContentHeight = NO;
+  if (_contentVCPreferredContentSizeHeightCached > preferredContentHeight) {
+    shouldSetContentHeight = YES;
+  } else {
+    if (_contentVCPreferredContentSizeHeightCached != preferredContentHeight) {
+      // Calculate height before we set preferredContentSize
+      CGFloat oldContentPreferredHeight = _contentVCPreferredContentSizeHeightCached;
+      CGFloat oldTotalHeight = [self totalHeightWithAddedContentHeight:oldContentPreferredHeight];
+      oldTotalHeight =
+          CGRectGetHeight(self.presentingViewBounds) - oldTotalHeight - MDCDeviceTopSafeAreaInset();
+      self.contentViewController.preferredContentSize = newPreferredContentSize;
```

Comment:
just a thought, are you able to use the new setContentOffset:animated: api here instead of re-calculating it again?